app/javascript/App/components/NewsfeedPage/CoinBody.js

```diff
@@ -1,64 +1,58 @@
-import React, { Component } from 'react'
+import React from 'react'
 import CoinCharts from '../CoinCharts'
 import Currency from '../Currency'
 import PercentageChange from '../PercentageChange'
-import WatchButton from '../WatchButton'
+import WatchButton from './WatchButton'
 import LoadingIndicator from '../LoadingIndicator'
 
-export default class CoinBody extends Component {
-  onWatchCoin = ({ id, watching }) => {
-    const details = { id, is_being_watched: watching }
-    this.props.setEntityDetails('coin', details)
-  }
-  render() {
-    const { selectCoinDetails, activeEntity, isLoading } = this.props
-    let coin = selectCoinDetails(activeEntity.id)
-    if (!coin && isLoading('coin'))
-      return <LoadingIndicator className="overlay bg-white-70" />
-    coin = coin.toJS()
-    return (
-      <div className="pa4">
-        <div className="flex justify-between items-center">
-          <div className="f4 fw6 flex items-center">
-            {coin.image_url && (
-              <img className="w2e h2e mr3" src={coin.image_url} alt="" />
-            )}
-            {coin.name}
-            <span className="ml2">({coin.symbol})</span>
-          </div>
-          <div>
-            <WatchButton
-              coinID={coin.id}
-              watching={coin.is_being_watched}
-              onChange={this.onWatchCoin}
-            />
-          </div>
+const CoinBody = (props) => {
+  const { selectCoinDetails, activeEntity, isLoading, user } = props
+  let coin = selectCoinDetails(activeEntity.id)
+  if (!coin && isLoading('coin'))
+    return <LoadingIndicator className="overlay bg-white-70" />
+  return (
+    <div className="pa4">
+      <div className="flex justify-between items-center">
+        <a
+          href={`/coins/${coin.get('slug')}`}
+          className="f4 fw6 flex items-center color-inherit"
+        >
+          {coin.get('image_url') && (
+            <img className="w2e h2e mr3" src={coin.get('image_url')} alt="" />
```

Comment:
this is just a comment: i don't feel it's worth the tradeoff, to adopt this framework that requires you add a bunch of abbreviated classes to do something simple. `className="w2e h2e mr3` just looks like greek, and the majority of people won't know what this means. to hit the ground running is impossible. they will need to have the API side by side and look up things. 

IMHO that in itself, is a good enough reason to avoid this like the plague. the vision is for the company to grow, and in that, the team will grow. think about what the community has adopted, rather than what has been successful for us. if we had jeremy ashkenas on our team, maybe we would get the most success using backbone. at the cost of making it harder for other people to work with and harder to attract talent. 

for the same reason we use react over ember. not that ember isn't well engineered, capable, or well documented. in fact many enterprise applications strongly evangelize ember. but it doesn't have a wide user base, and is harder to find ember developers. in the same vein, it's harder to find tachyon developers. they are a small subset of the css ecosystem.

Reply:
I'm a big fan of functional CSS frameworks like Tachyons or BassCSS, and the learning curve isn't steep I think. What I do is keep the table of styles open in a tab and search for the utility I need.
This page makes a good case: https://github.com/dwyl/learn-tachyons#30-second-summary

I'd be open to using something like Styled Components, but only because it would indeed allow other devs to get started more quickly.

Reply:
@cryptoB0T For what it's worth, I started off a disbeliever but kept an open mind, and less than a week into development I was already reasonably fluent in Tachyons.

It's a lot more flexible than something like Bootstrap and the learning curve is pretty shallow - although the beginning feels very "Greek" in that it's a jumble of letters, the paradigm is quite straightforward to pick up and the versatility started feeling very powerful.

Think of it like `vim` for frontend design; the initial learning curve sucks but once you understand the paradigm, it's pretty cool to be able to do compositions and the expressiveness is much nicer than other more rigid frontend frameworks.